app/src/main/res/layout/main_fragment.xml

```diff
@@ -308,7 +308,7 @@
             app:backgroundTint="@color/creamBackground" />
 
         <Button
-            android:id="@+id/button3"
+            android:id="@+id/getRandomPuzzleButton"
             style="@style/Widget.AppCompat.Button.Borderless"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
```

Comment:
I renamed `button2` to `newPuzzleButton` and `button3` to `getRandomPuzzleButton` as these names make more sense and make it easier to reference in the future. I followed the kotlin coding convention's "choose good names" rule which describes how each entity should have a meaningful name to it. Following this change, I had to refactor the usage of the buttons as well. Reference - https://kotlinlang.org/docs/coding-conventions.html#choose-good-names

Reply:
Good.